diff --git a/src/Makefile b/src/Makefile
index f32d2ca..f5ee3d8 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -29,7 +29,7 @@ LLVM_STRIP ?= llvm-strip
 BPFTOOL ?= bpftool
 BPF_INCLUDE := /usr/include
 BPF_LOCAL_INCLUDE := /usr/local/include
-BPF_CFLAGS := -g -fno-stack-protector -Wall -Wno-incompatible-pointer-types-discards-qualifiers
+BPF_CFLAGS := -g -fno-stack-protector -Wno-incompatible-pointer-types-discards-qualifiers
 NL_INCLUDE := /usr/include/libnl3
 INCLUDES := -I../include -I$(BPF_INCLUDE) -I$(BPF_LOCAL_INCLUDE) -I$(NL_INCLUDE) -I../include/uapi
 INSTALL ?= install
@@ -59,7 +59,7 @@ endif
 VERSION = 0.2.1
 VERSION_SCRIPT  := libbpftune.map
 
-CFLAGS = -fPIC -Wall -Wextra -g -I../include -std=c99
+CFLAGS = -fPIC -Wextra -g -I../include -std=c99
 
 CFLAGS += -DBPFTUNE_VERSION='"$(BPFTUNE_VERSION)"' \
 	  -DBPFTUNER_PREFIX_DIR='"$(prefix)"' \
@@ -123,18 +123,12 @@ BPFTUNE_HDRS = ../include/bpftune/libbpftune.h \
 
 .PHONY: clean
 
-ifeq ($(GCC_BPF),)
-all: analyze $(OPATH) $(OPATH)bpftune $(TUNER_LIBS)
-else
 BPF_CFLAGS += -std=gnu17
 all: $(OPATH) $(OPATH)bpftune $(TUNER_LIBS)
-endif
 
 $(OPATH):
 	mkdir $(OPATH)
 
-analyze: $(BPF_SKELS) $(LEGACY_BPF_SKELS) $(NOBTF_BPF_SKELS)
-	$(CLANG) --analyze $(INCLUDES) libbpftune.c bpftune.c $(TUNER_SRCS)
 clean:
 	$(call QUIET_CLEAN, bpftune)
 	$(Q)$(RM) $(OPATH)*.o *.d $(OPATH)*.so*
