diff --git a/installkernel b/installkernel
index 7e65405..f39a815 100644
--- a/installkernel
+++ b/installkernel
@@ -83,7 +83,7 @@ fi
 # If installing in the usual directory, run the same scripts that hook
 # into kernel package installation.  Also make sure the PATH includes
 # /usr/sbin and /sbin, just as dpkg would.
-if [ "$dir" = "/boot" ]; then
+if [[ -d /etc/kernel/postinst.d && "$dir" = "/boot" ]]; then
   PATH="$PATH:/usr/sbin:/sbin" \
     run-parts --verbose --exit-on-error --arg="$ver" --arg="$dir/$img_dest-$ver" \
     /etc/kernel/postinst.d
